modules/js-and-css/speculation-rules/class-plsr-url-pattern-prefixer.php

```diff
@@ -86,8 +86,32 @@ public function prefix_path_pattern( string $path_pattern, string $context = 'ho
 	 */
 	public static function get_default_contexts(): array {
 		return array(
-			'home' => trailingslashit( wp_parse_url( home_url( '/' ), PHP_URL_PATH ) ),
-			'site' => trailingslashit( wp_parse_url( site_url( '/' ), PHP_URL_PATH ) ),
+			'home' => self::escape_pattern_string( trailingslashit( wp_parse_url( home_url( '/' ), PHP_URL_PATH ) ) ),
+			'site' => self::escape_pattern_string( trailingslashit( wp_parse_url( site_url( '/' ), PHP_URL_PATH ) ) ),
 		);
 	}
+
+	/**
+	 * Escapes a string for use in a URL pattern component.
+	 * https://urlpattern.spec.whatwg.org/#escape-a-pattern-string
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $str String to be escaped.
+	 * @return string String with backslashes added where required.
+	 */
+	public static function escape_pattern_string( string $str ): string {
```

Comment:
Right now this method is only used within this class. If it remains like that, it should probably become private.